packages/indexer/src/sinks/csv.ts

```diff
@@ -0,0 +1,95 @@
+import type { Cursor } from "@apibara/protocol";
+import { Sink, type SinkWriteArgs } from "../sink";
+import { type Stringifier, stringify, type Options } from "csv-stringify";
+import fs from "node:fs";
+
+export type CsvArgs = {
+  /**
+   * csv-stringy options
+   * @reference https://csv.js.org/stringify/options/
+   */
+  csvOptions?: Options;
+  /**
+   * filepath for your csv file
+   */
+  filepath: string;
+};
+
+export type CsvSinkOptions = {
+  /**
+   * An optional column name used to store the cursor value. If specified,
+   * the value of this column must match the `endCursor.orderKey` for each row.
+   */
+  cursorColumn?: string;
+};
+
+class CsvSink<TData extends Record<string, unknown>> extends Sink<TData> {
+  private _config: CsvSinkOptions;
+  private _stringifier: Stringifier;
+
+  constructor(stringifier: Stringifier, config: CsvSinkOptions) {
+    super();
+    this._config = config;
+    this._stringifier = stringifier;
+  }
+
+  async write({ data, endCursor }: SinkWriteArgs<TData>) {
+    this.emit("write", { data });
+    // adds a "_cursor" property if "cursorColumn" is not specified by user
+    data = this.processCursorColumn(data, endCursor);
```

Comment:
Let's not do it now, but when we will add the remaining sinks we should have an "helpers" module with a function that adds the curser column to the rows.